nanoprintf.h

```diff
@@ -241,27 +242,24 @@ typedef enum {
   NPF_FMT_SPEC_CONV_FLOAT_SHORTEST, // 'g', 'G'
   NPF_FMT_SPEC_CONV_FLOAT_HEX,      // 'a', 'A'
 #endif
-} npf_format_spec_conversion_t;
+};
 
 typedef struct npf_format_spec {
-  char prepend;          // ' ' or '+'
-  char alt_form;         // '#'
-
 #if NANOPRINTF_USE_FIELD_WIDTH_FORMAT_SPECIFIERS == 1
-  npf_fmt_spec_opt_t field_width_opt;
   int field_width;
+  uint8_t field_width_opt;
   char left_justified;   // '-'
   char leading_zero_pad; // '0'
 #endif
-
 #if NANOPRINTF_USE_PRECISION_FORMAT_SPECIFIERS == 1
-  npf_fmt_spec_opt_t prec_opt;
   int prec;
+  uint8_t prec_opt;
 #endif
-
-  npf_format_spec_length_modifier_t length_modifier;
-  npf_format_spec_conversion_t conv_spec;
-  char case_adjust;
+  char prepend;          // ' ' or '+'
+  char alt_form;         // '#'
+  char case_adjust;      // 'a' - 'A'
+  uint8_t length_modifier;
+  uint8_t conv_spec;
```

Comment:
Are there any wins from these becoming fast types?

Reply:
Well, the bottom two, I guess. The top 3 are used exclusively in addition / comparison with other char-sized types; I remember that was a win because it saved some mask instructions.

Reply:
I tried this:
```
typedef struct npf_format_spec {
#if NANOPRINTF_USE_FIELD_WIDTH_FORMAT_SPECIFIERS == 1
  int field_width;
  uint_fast8_t field_width_opt;
  char left_justified;   // '-'
  char leading_zero_pad; // '0'
#endif
#if NANOPRINTF_USE_PRECISION_FORMAT_SPECIFIERS == 1
  int prec;
  uint_fast8_t prec_opt;
#endif
  uint_fast8_t length_modifier;
  char prepend;          // ' ' or '+'
  char alt_form;         // '#'
  char case_adjust;      // 'a' - 'A'
  uint8_t conv_spec;
} npf_format_spec_t;
```
Saves 16 B for CM7 and 2 B for RL78 binary size. No change for CM0, CM3, PIC32, Xtensa, Quark, MSP430, AVR. Increases the struct/stack size by 8 B for Cortex and probably other 32-bit CPUs. In addition to that, changing the type of the `conv_spec` element to fast type ends up mostly increasing the size. Go figure...